filebeat/harvester/harvester.go

```diff
@@ -39,6 +40,8 @@ type Harvester struct {
 	fileReader      *LogFile
 	encodingFactory encoding.EncodingFactory
 	encoding        encoding.Encoding
+	once            sync.Once
```

Comment:
it should be clear `once` is used to protected `prospectorDone` from multiple go-routines potentially doing `close(prospectorDone)`. Uhm.. is it used for `prospectorDone`?

Just seeing it protects the `done` channel. sync tools and items being protected should always stand together.